crates/types/src/schema/metadata/updater/mod.rs

```diff
@@ -354,29 +355,39 @@ impl SchemaUpdater {
             .collect::<Result<HashMap<_, _>, _>>()?;
 
         // Did we find an existing deployment with a conflicting endpoint url?
-        let mut existing_deployments = self.schema.deployments.iter().filter(|(_, schemas)| {
-            schemas.ty.protocol_type() == deployment_metadata.ty.protocol_type()
-                && schemas.ty.normalized_address() == deployment_metadata.ty.normalized_address()
-        });
+        let existing_deployment = self
+            .schema
+            .deployments
+            .iter()
+            .filter(|(_, schemas)| {
+                schemas.ty.protocol_type() == deployment_metadata.ty.protocol_type()
+                    && schemas.ty.normalized_address()
+                        == deployment_metadata.ty.normalized_address()
+                    && routing_header.as_ref().is_none_or(
+                        |(routing_header_key, routing_header_value)| {
+                            schemas
+                                .delivery_options
+                                .additional_headers
+                                .get(routing_header_key)
+                                .is_some_and(|v| v == routing_header_value)
+                        },
+                    )
+            })
+            // There are few situations where we might have multiple deployments for the same endpoint:
+            // * If the user specified in the at least two previous registrations the routing-header,
+            //   but then it didn't specify it in this one. In this case, multiple deployments will match the above filter
```

Comment:
I am not sure I get this correctly. I would like to think of the routing header as part of the deployment address. In that case, if no routing header is provided it means this can conflict with max of 1 deployment.

Reply:
>  In that case, if no routing header is provided it means this can conflict with max of 1 deployment.

This is coming directly from above, and can happen when you have the following sequence:

* localhost:9080 with routing header 1
* localhost:9080 with routing header 2
* localhost:9080 without routing header -> boom, this condition with multiple deployments kicks in